### PR TITLE
chore: Create home screen bottom sheet prototype

### LINKED ISSUE #12: feat: Create bottom sheet view for home screen
A swipe-abale cards view of the user's current requests needs to be implemented for the bottom sheet of the home screen. This entails creating a bottom sheet that has a preview and can be swiped up (while still leaving space at the top). Within this bottom sheet will be a horizontally swipe-abale list of cards, each containing the details for a bulletin the user is currently signed up for.

### PR DESCRIPTION
**Pull request by Neha-Rode**
# Description of Changes
Resolves #12 

Created a bottom sheet view to hold request details for the home sheet. This includes the creation of a `RequestDetailsCard()`, which is the contents of the bottom modal sheet. Also imported a  `DotsIndicator` package to help index the cards in the bottom sheet. 

# Other Notes
The status indicator widget was updated to include a `hideBottomText` parameter, which allows the developer to hide the label text from the status indicator. Such an instance of this widget is used in the preview for the bottom sheet in the home screen. Also created a `RequestDetails()` widget for convenience, since it is used in both the `RequestDetailsCard()` and the `RequestDetailsScreen()` widgets.

### CODE REVIEW

**Comment by Neha-Rode** [comment]
While creating the widget, it did occur to me that the ability to scroll vertically within a card will be confusing to work with, since the entire bottom sheet itself is scrollable. I believe we discussed this during the design progress, but dismissed it for the time being. 

While I think this may be fine for now, I think it might be worth considering to eventually have the bottom sheet become a full screen view upon expansion. This way, I think it would make more intuitive to have scrollable content within the bottom sheet. I believe the current version of GoogleMaps implements something similar to this (when you tap the "Explore nearby" preview of the bottom sheet).

File: lib/ui/theme/theme.dart [hunk 1]
```diff
@@ -12,67 +12,67 @@ class ConcordThemeManager with ChangeNotifier {
...
     overline: TextStyle(
       fontFamily: 'Lato',
-      fontSize: 10,
+      fontSize: 12,
       fontWeight: FontWeight.w500,
     ),
     caption: TextStyle(
       fontFamily: 'Lato',
-      fontSize: 12,
+      fontSize: 14,
```
Comment (WillieCubed) [inline lib/ui/theme/theme.dart:75]: Is there a reason the font sizes have been changed from the design mockups and type system we adopted from the [Material Design guidelines](https://material.io/design/typography/the-type-system.html#type-scale)?

File: lib/ui/screens/home_screen.dart [hunk 2]
```diff
@@ -1,24 +1,203 @@
...
+              /// When the user swipes on the card, [itemIndex] is updated so that
+              /// the DotsIndicator can indicate the change in page view.
+              onPageChanged: (int index){
+                setState(() {
+                  itemIndex = index.toDouble();
+                });
+              },
+            ),
+          ),
+          DotsIndicator(
```
Comment (WillieCubed) [inline lib/ui/screens/home_screen.dart:186]: Can you add the counter below this as noted in the mockup?

File: lib/ui/theme/theme.dart (hunk 1 above)
Comment (Neha-Rode) [reply lib/ui/theme/theme.dart:75]: Everything just looked weirdly small on the emulator, but like if those sizes were literally from the guidelines, I'm just gonna change them back hehe oops.

File: lib/ui/screens/home_screen.dart (hunk 2 above)
Comment (Neha-Rode) [reply lib/ui/screens/home_screen.dart:186]: Yes of course sir.

**Pull request merged**